lib/mustache_context.dart

```diff
@@ -124,12 +124,30 @@ class _MustacheContext extends MustacheToString implements MustacheContext {
   }
 
   dynamic _getActualValue(String key) {
-    try {
-      return ctx[key];
-    } catch (NoSuchMethodError) {
-      //Try to make dart2js understand that when we define USE_MIRRORS = false
-      //we do not want to use any reflector
-      return (useMirrors && USE_MIRRORS) ? ctxReflector[key] : null;
+    //Try to make dart2js understand that when we define USE_MIRRORS = false
+    //we do not want to use any reflector as that inflates the generated
+    //javascript.
+    if (useMirrors && USE_MIRRORS) {
+      if (reflect(ctx).type.instanceMembers.containsKey(new Symbol("[]"))) {
+        MethodMirror m = reflect(ctx).type.instanceMembers[new Symbol("[]")];
+        TypeMirror reflectedString = reflectType(String);
+        if (reflectedString.isAssignableTo(m.parameters[0].type)) {
+          try {
+            return ctx[key];
+          } catch (NoSuchMethodError) {
```

Comment:
Previously, if (useMirrors && USE_MIRRORS) is true and the attempt to use ctx[key] resulted in an exception, we returned ctxReflector[key].  So a noop here falls through to the base case of returning ctxReflector[key].  The other two if statements just try to get us to return ctxReflector[key] if we can predict in advance that calling ctx[key] will raise an exception.